policy/release/trusted_task.rego

```diff
@@ -145,6 +147,37 @@ deny contains result if {
 	result := lib.result_helper(rego.metadata.chain(), [])
 }
 
+# METADATA
+# title: Trusted parameters
+# description: >-
+#   Confirm certain parameters provided to each builder Task have come from trusted Tasks.
```

Comment:
I think we want to check every Task on the chain not just the builder Tasks. Perhaps one can manipulate the prefetch-dependencies Task via a parameter from an untrusted Task? Not really via digests as it currently stands, but still...

Reply:
Let me ponder on this for a bit. I think if someone is using Trusted Artifacts, it's ok if they have certain Tasks that do this as long as it doesn't affect the builder Tasks.

Reply:
Here is an example, I add an untrusted Task `pick-proxy` which will emit a `HTTP_PROXY` result, i.e. a HTTP proxy that will be used for the git-clone or prefetch-dependencies Task, now in good faith this Task was added to choose a correct proxy, say depending on where the build is running. If that Task is compromised then the sources or the dependencies can be influenced. That's why I think we should look at the whole chain. This is in the end similar to how the non Trusted Artifacts pipeline is examined where we say all Tasks need to be trusted.

Reply:
It's a valid point. I just don't think we have enough information in the SLSA Provenance to do this. We need the Pipeline definition. I'll file a follow up spike for this.

Reply:
Filed https://issues.redhat.com/browse/EC-670